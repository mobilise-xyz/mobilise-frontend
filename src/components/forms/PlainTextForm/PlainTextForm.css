textarea {
    height: 1.5rem !important;
}

textarea#description.form-control {
    height: 3rem !important;
}