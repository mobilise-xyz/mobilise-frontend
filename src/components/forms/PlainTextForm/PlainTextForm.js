import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';
import './PlainTextForm.css';

const PlainTextForm = props => {
  const {
    noLabel = false,
    label = '',
    name,
    placeholder = '',
    content,
    disabled = false,
    handleChange
  } = props;
  return (
    <Form.Group>
      {noLabel ? null : (
        <Form.Label>
          {label.charAt(0).toUpperCase() + label.slice(1)}
        </Form.Label>
      )}
      <Form.Control
        as="textarea"
        id={label}
        name={name}
        disabled={disabled}
        placeholder={placeholder}
        value={content}
        onChange={handleChange}
        required
      />
    </Form.Group>
  );
};

PlainTextForm.defaultProps = {
  noLabel: false,
  label: '',
  placeholder: '',
  disabled: false
};

PlainTextForm.propTypes = {
  noLabel: PropTypes.bool,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  content: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  handleChange: PropTypes.func.isRequired
};

export default PlainTextForm;
