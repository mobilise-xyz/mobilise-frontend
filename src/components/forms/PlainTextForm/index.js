import PlainTextForm from './PlainTextForm';

export default PlainTextForm;
