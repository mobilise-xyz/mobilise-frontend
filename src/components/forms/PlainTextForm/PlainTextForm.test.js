import React from 'react';
import renderer from 'react-test-renderer';
import PlainTextForm from './PlainTextForm';

describe('Description Form Component', () => {
  it('renders the form correctly', () => {
    const component = renderer.create(
      <PlainTextForm label="" handleChange={() => {}} content="" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
