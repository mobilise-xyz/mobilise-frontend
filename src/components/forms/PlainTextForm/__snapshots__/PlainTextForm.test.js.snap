// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Form Component renders the form correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    
  </label>
  <textarea
    className="form-control"
    disabled={false}
    onChange={[Function]}
    placeholder=""
    required={true}
    value=""
  />
</div>
`;
